common/appointment/get.ts

```diff
@@ -0,0 +1,70 @@
+import { Lecture as Appointment } from '../entity/Lecture';
+import { getUserType, User } from '../user';
+import { prisma } from '../prisma';
+import { AttendanceStatus } from '../entity/AppointmentAttendee';
+
+export const getAppointmentsForUser = async (user: User, take?: number, skip?: number): Promise<Appointment[]> => {
+    const userType = getUserType(user);
+    let appointmentIds: number[] = [];
+    switch (userType) {
+        case 'student':
+            appointmentIds = await getStudentAppointmentIds(user.studentId);
+            break;
+        case 'pupil':
+            appointmentIds = await getPupilAppointmentIds(user.pupilId);
+            break;
+        case 'screener':
+            appointmentIds = await getScreenerAppointmentIds(user.screenerId);
+            break;
+        default:
+            throw new Error(`Unknown user type: ${userType}`);
+    }
+    appointmentIds = deduplicate(sortIds(appointmentIds));
```

Comment:
sortIds(...) is a no op as we only use it to select from the DB anyways, and there we ORDER BY start date ...

Reply:
deduplicate too technically